### PR TITLE
Add items to Quantum RPG

### PR DESCRIPTION
**Pull request by dstrain115**
- Add items you can interact with in Quantum RPG.
- These will be useful for signs, helpful NPCS, and puzzles.
- Right now, they can only print out a string.

Follow-up PRs will hook in custom functions for more complicated puzzles.  For right now, there is a untyped 'Callable' as a placeholder for such customizations until I figure out what the function signature should be (or maybe it will be a class, like ItemAction).

### CODE REVIEW

File: unitary/examples/quantum_rpg/main_loop_test.py [hunk 1]
```diff
@@ -77,7 +84,7 @@ def test_do_simple_move() -> None:
     output = io.StringIO()
     c = classes.Analyst("Mensing")
     loop = main_loop.MainLoop([c], world.World(EXAMPLE_WORLD), output)
-    loop.loop(user_input=["e", "w", "quit"])
+    loop.loop(user_input=["e", "read sign", "w", "quit"])
```
Comment (losos0) [inline unitary/examples/quantum_rpg/main_loop_test.py:87]: The second word in "read sign" intentionally doesn't matter?

File: unitary/examples/quantum_rpg/main_loop.py [hunk 2]
```diff
@@ -92,6 +97,14 @@ def loop(self, user_input: Optional[Sequence[str]] = None) -> None:
             if cmd is not None:
                 self.world.move(cmd)
                 continue
+            words = current_input.split()
+            action = self.world.current_location.get_action(words[0])
```
Comment (losos0) [inline unitary/examples/quantum_rpg/main_loop.py:101]: This appears to throw on an empty line

Comment (dstrain115) [reply unitary/examples/quantum_rpg/main_loop.py]: Removed.

Comment (losos0) [reply unitary/examples/quantum_rpg/main_loop.py]: There is still the same problem; the code that throws on an empty line is just in another function.

LGTM other than that

Comment (dstrain115) [reply unitary/examples/quantum_rpg/main_loop.py]: Fixed, with test.  I misunderstood what you meant at first.

**Pull request merged**